extension/src/commands.ts

```diff
@@ -39,5 +41,56 @@ export function registerCommands(context: vscode.ExtensionContext){
 	}));
 
 
+	// Associated with the right click "Generate Documentation" menu item
+	context.subscriptions.push(vscode.commands.registerCommand('autolisp.generateDocumentation', async () => {
+		try {
+			const pos = vscode.window.activeTextEditor.selection.start;
+			const doc = AutoLispExt.Documents.getDocument(vscode.window.activeTextEditor.document);
+			const exp = doc.atomsForest.find(p => p.contains(pos));
+
+			// Locate all the Defun & Defun-Q statements or an empty array if no actionable scope was found
```

Comment:
One comment code block is one "small" function, maybe we can refractor it to more readable.